src/template_builder.py

```diff
@@ -42,90 +55,31 @@ def read_template_parts_list(template_parts_list_file):
-            field['group_id'] = new_id
-
-        return template_parts_file_content
-
-    @staticmethod
-    def concatenate(existing_template_parts_content,
-                    new_template_part_content):
-        """
-        Concatenates two template part contents.
-
-        Parameters:
-            existing_template_parts_content (dict): Existing content.
-            new_template_part_content (dict): New content to add.
-
-        Returns:
-            dict: Merged content.
-        """
-        existing_template_parts_content['elabftw'][
-            'extra_fields_groups'].extend(
-            new_template_part_content['elabftw']['extra_fields_groups']
-        )
-        existing_template_parts_content['extra_fields'].update(
-            new_template_part_content['extra_fields']
-        )
-        return existing_template_parts_content
-
-    @staticmethod
-    def save_template(full_template_content, template_file_path):
+    def build_template(template_parts_list_file, template_file_path):
         """
-        Saves the template content to a JSON file.
+     Generates a complete template by merging all the listed parts.
 
-        Parameters:
-            full_template_content (dict): Content to save.
-            template_file_path (str): Path to the output file.
+     Parameters:
+         template_parts_list_file (str): Path to the CSV file listing the parts.
+         template_file_path (str): Path to the output JSON file.
 
-        Returns:
-            None
-        """
-        with open(template_file_path, 'w') as f:
-            json.dump(full_template_content, f, indent=4)
+     Returns:
+         None
+     """
+        template_builder = TemplateBuilder(template_parts_list_file)
 
-    @staticmethod
-    def generate_template(template_parts_list_file, template_file_path):
-        """
-        Generates a complete template by merging all listed parts.
-
-        Parameters: template_parts_list_file (str): Path to the CSV file
-        listing the parts. template_file_path (str): Path to the output JSON
-        file.
-
-        Returns:
-            None
-        """
-        template_parts_list = TemplateBuilder.read_template_parts_list(
-            template_parts_list_file)
-        full_template_content = None
+        full_template = None
 
-        for new_id, template_part_file in enumerate(template_parts_list):
+        for new_id, template_part_file in enumerate(
+                template_builder.template_parts_list):
             template_part = TemplatePart(template_part_file)
-            new_template_part_content = TemplateBuilder.set_content_id(
-                new_id + 1, template_part.template_content)
 
-            if full_template_content is None:
-                full_template_content = new_template_part_content
+            new_template_part = template_part.set_content_id(new_id + 1)
+
+            if full_template is None:
```

Comment:
testing is full_template is None like this is a bit cheating ;)

you should use the constructor (cf comment above) to start from an empty Template, and here test with a method Template.is_empty() that you should implement... that would be way better!